### PR TITLE
Add code coverage to CI

### PR DESCRIPTION
**Pull request by mmarchini**
Add support to gcov through a `npm install` flag
(`--llnode_coverage=true`), and update Travis settings to upload
coverage report to CodeCov on success.

### CODE REVIEW

**Comment by mmarchini** [comment]
Oh cool, seems like we don't need to set up anything in the repository 😄

**Comment by mmarchini** [comment]
I added code coverage report for our tests as well (with `nyc`):

image

**Comment by mmarchini** [comment]
cc @joyeecheung @cjihrig

File: scripts/configure.js [hunk 1]
```diff
@@ -19,15 +19,24 @@ function main() {
...
   const build_addon = (process.env.npm_config_llnode_build_addon ||
     process.env.LLNODE_BUILD_ADDON);
   if (build_addon) {
     config.variables.build_addon = build_addon;
   } else {
     config.variables.build_addon = 'false';
   }
+
+  // Optionally build with support for gcov
+  const coverage = (process.env.npm_config_llnode_coverage ||
```
Comment (cjihrig) [inline scripts/configure.js]: Unnecessary parens.

**Comment by mmarchini** [comment]
Landed in 59b65c4...9f3dce3

**Pull request closed without merging**